server.go

```diff
@@ -102,6 +104,14 @@ func doDecrypt(ctx context.Context, data string) ([]byte, error) {
 }
 
 func main() {
+	shutdown := make(chan int)
```

Comment:
a single `shutdown` channel will do. We don't need to listen on one then pass thru to the other.